lighthouse-core/audits/service-worker.js

```diff
@@ -73,23 +73,31 @@ class ServiceWorker extends Audit {
    * @param {Array<LH.Crdp.ServiceWorker.ServiceWorkerVersion>} matchingSWVersions
    * @param {Array<LH.Crdp.ServiceWorker.ServiceWorkerRegistration>} registrations
    * @param {URL} pageUrl
-   * @return {string|undefined}
+   * @return {Array<string> | undefined}
    */
   static getControllingScopeUrl(matchingSWVersions, registrations, pageUrl) {
     // Find the normalized scope URLs of possibly-controlling SWs.
-    const matchingScopeUrls = matchingSWVersions
-      .map(v => registrations.find(r => r.registrationId === v.registrationId))
-      .filter(/** @return {r is LH.Crdp.ServiceWorker.ServiceWorkerRegistration} */ r => !!r)
-      .map(r => new URL(r.scopeURL).href);
+    const serviceWorkerUrls = new Map();
+
+    // Populate serviceWorkerUrls map with the scriptURLs and scopeUrls of matchingSWVersions and registrations
+    matchingSWVersions.forEach(function(version) {
+      const tempRegistration = registrations.find(r => r.registrationId === version.registrationId);
+      if (tempRegistration) {
+        const scopeURL = new URL(tempRegistration.scopeURL).href;
+        const scriptURL = new URL(version.scriptURL).href;
+        serviceWorkerUrls.set(scopeURL, scriptURL);
+      }
+    });
 
     // Find most-specific applicable scope, the one controlling the page.
     // See https://w3c.github.io/ServiceWorker/v1/#scope-match-algorithm
-    const pageControllingScope = matchingScopeUrls
-      .filter(scopeUrl => pageUrl.href.startsWith(scopeUrl))
-      .sort((scopeA, scopeB) => scopeA.length - scopeB.length)
+    const pageControllingUrls = [...serviceWorkerUrls]
+      /* converts map to array, to properly filer and sort according to scopeUrl */
```

Comment:
using a Map() is good, but right now it pretty quickly is thrown away and we turn it into the `[['scope1', 'scriptA'],['scope2','scriptB']]` representation. So we don't get much value out of it.

and separately the current return value is an array of strings, but you kinda have to know which is which.. I'd feel better making it explicit.


i remember in an earlier version you had named properties. i think within the `matchingScopeUrls` functional methods you ended up making a `Array<{scopeUrl: string, scriptUrl: string}>` or something

@adrianaixba  you were right and that was better. :) i misled you as i thought doing that wasn't necessary. mea cupla.

so yeah  looking at it now.. the best return value for this `getControllingServiceWorker()` method would probably be `{{scopeUrl: string, scriptUrl: string}}`, so we might as well build that from the start.